addons/website_event/models/event_event.py

```diff
@@ -88,19 +92,19 @@ def _get_menu_entries(self):
             (_('Register'), '/event/%s/register' % slug(self), False),
         ]
 
-    def _update_website_menus(self, vals):
+    def _update_website_menus(self):
         for event in self:
-            if 'website_menu' in vals:
-                if event.menu_id and not event.website_menu:
-                    event.menu_id.unlink()
-                elif event.website_menu and not event.menu_id:
-                    root_menu = self.env['website.menu'].create({'name': event.name, 'website_id': event.website_id.id})
-                    event.menu_id = root_menu
-                    for sequence, (name, url, xml_id) in enumerate(event._get_menu_entries()):
-                        event._create_menu(sequence, name, url, xml_id)
+            if event.menu_id and not event.website_menu:
+                event.menu_id.unlink()
+            elif event.website_menu and not event.menu_id:
+                root_menu = self.env['website.menu'].create({'name': event.name, 'website_id': event.website_id.id})
+                event.menu_id = root_menu
+                for sequence, (name, url, xml_id) in enumerate(event._get_menu_entries()):
+                    event._create_menu(sequence, name, url, xml_id)
 
     def _create_menu(self, sequence, name, url, xml_id):
         if not url:
+            self.env['ir.ui.view'].search([('name', '=', name + ' ' + self.name)]).unlink()
```

Comment:
As self.name is translatable, it will probably fail.

Moreover why do you remove a view here ?

Reply:
Hello @tde-banana-odoo 

Right now, when disabling the website menu and enabling it again after it creates another ir.ui.view as the new_page(ispage=False) method is called. Therefore you get a traceback when clicking on the customize dropdown on the frontend. It's the only way I found to unlink that ir.ui.view.

Reply:
Indeed it is a bit complicated as I did not find any link to the view ... broken by design.

Ok then, to be cleaned in master post freeze with real links.